Lib/multiprocessing/pool.py

```diff
@@ -173,7 +173,16 @@ def __init__(self, processes=None, initializer=None, initargs=(),
 
         self._processes = processes
         self._pool = []
-        self._repopulate_pool()
+        try:
+            self._repopulate_pool()
+        except OSError:
+            for p in self._pool:
+                if p.exitcode is None:
+                    p.terminate()
+            for p in self._pool:
+                if p.is_alive():
```

Comment:
I don't think you need the process to be alive to join it.

Reply:
Look ok not to test it first.